astropy/io/misc/asdf/tags/helpers.py

```diff
@@ -1,11 +1,16 @@
 # Licensed under a 3-clause BSD style license - see LICENSE.rst
 # -*- coding: utf-8 -*-
 
-import numpy as np
+from pkg_resources import get_distribution
 
+import numpy as np
 
 __all__ = []
 
+for cur_req in get_distribution('astropy').requires(extras=('all', )):
```

Comment:
If we make this a little helper function in `utils/introspection`, we could do a python-version dependent choice of `pkg_resources` or `importlib.metadata` and thus future-proof ourselves.

It can also raise an error if the name is not found (and perhaps we can use it in `conf.py` as well?)

Reply:
Re: this and https://github.com/astropy/astropy/issues/11059#issuecomment-732393971 -- I am not sold on a function in `utils` to enable switching between `pkg_resources` and `importlib.metadata`, at least not for this PR.

Notable differences between the two:

* Using `pkg_resources`, I have to extract Python minversion from `setup.cfg`, which isn't available in the installed version. For `importlib.metadata`, it is given.
* Using `pkg_resources`, the minversion is parsed for you in `.specs` attribute. In `importlib.metadata`, it is given in a string (e.g., `"PyYAML (>=3.13) ; extra == 'docs'"`) that you then have to parse yourselves using regex because the `(>=3.13)` part can be missing if no minversion is specified.

I also don't see much use of this function post-installation, so why maintain one more public API?

Reply:
One option with `importlib.metadata` :
```
In [31]: import importlib.metadata
    ...: from packaging.requirements import Requirement
    ...: 
    ...: versions = {}
    ...: for line in importlib.metadata.requires('astropy'):
    ...:     req = Requirement(line.split(';')[0])
    ...:     versions[req.name] = str(req.specifier)
    ...: 

In [32]: versions
Out[32]: 
{'numpy': '>=1.17',
 'pyerfa': '',
 'scipy': '>=1.1',
 'dask': '',
 'h5py': '',
 'beautifulsoup4': '',
 'html5lib': '',
 'bleach': '',
 'PyYAML': '>=3.13',
 'pandas': '',
 'sortedcontainers': '',
 'pytz': '',
 'jplephem': '',
 'matplotlib': '>=3.1',
 'mpmath': '',
 'asdf': '>=2.6',
 'bottleneck': '',
 'ipython': '',
 'pytest': '',
 'sphinx': '',
 'sphinx-astropy': '>=1.3',
 'pytest-astropy': '>=0.8',
 'pytest-xdist': '',
 'objgraph': '',
 'coverage': '',
 'skyfield': '>=1.20',
 'sgp4': '>=2.3'}
```